static/app/components/searchSyntax/parser.tsx

```diff
@@ -671,6 +681,25 @@ export class TokenConverter {
     return null;
   };
 
+  /**
+   * Checks the validity of a logical boolean filter based on the provided search configuration
+   */
+  checkInvalidLogicalBoolean = (value: BooleanOperator) => {
+    if (this.config.disallowedLogicalOperators.has(value)) {
+      return value === BooleanOperator.OR
+        ? {
+            type: InvalidReason.LOGICAL_OR_NOT_ALLOWED,
+            reason: this.config.invalidMessages[InvalidReason.LOGICAL_OR_NOT_ALLOWED],
+          }
+        : {
+            type: InvalidReason.LOGICAL_AND_NOT_ALLOWED,
+            reason: this.config.invalidMessages[InvalidReason.LOGICAL_AND_NOT_ALLOWED],
+          };
+    }
```

Comment:
Should we be explicit with this conditional actually, not saying there would ever be more than a `OR` and `AND`, but it might just be a bit more clear to have an exhaustive if check type thing
```suggestion
      if (value === BooleanOperator.OR) {
        return {
          type: InvalidReason.LOGICAL_OR_NOT_ALLOWED,
          reason: this.config.invalidMessages[InvalidReason.LOGICAL_OR_NOT_ALLOWED],
        };
      }
      if (value === BooleanOperator.AND) {
        return {
          type: InvalidReason.LOGICAL_AND_NOT_ALLOWED,
          reason: this.config.invalidMessages[InvalidReason.LOGICAL_AND_NOT_ALLOWED],
        };
      }
    }
```